### ISSUE TITLE
Pull request is in unstable status

### ISSUE THREAD

**Issue by izavads** [peter-evans/enable-pull-request-automerge#355]
Hi , 
getting the below message when trying to use the action : 

Request failed due to following response errors: - ["Pull request Pull request is in unstable status"]

```
      - uses: peter-evans/enable-pull-request-automerge@v2
        with:
          token: ${{ secrets.MERGE_SECRET }}
          pull-request-number: ${{ github.event.number }}
          merge-method: squash
```

**Comment by novotl** [comment]
👍 We're observing the same issue

**Comment by Celebrate-Reinhard** [comment]
same error here

**Comment by peter-evans** [comment]
Since you all seemed to experience this at the same time, it's likely that it's an issue on GitHub's side. The action itself has not changed in any way.

**Comment by Celebrate-Reinhard** [comment]
@peter-evans Thanks for the quick answer!
In my case: I try to use your (great) action for just some days now. So I cannot ensure that it worked before.

@novotl @izavads Did it work for you before?

**Comment by peter-evans** [comment]
Ah, my misunderstanding. In that case, please check that you are fulfilling the conditions described here:
https://github.com/peter-evans/enable-pull-request-automerge#conditions

**Comment by Celebrate-Reinhard** [comment]
@izavads 

> Once moved the automerge job to be the first job without any dependency it works as expected .

Can you give more details how exactly you did it?

I tried the same and split them up into two different workflows. -> Same error:

While some checks are still running (and therefor the auto-merge requirements are not satisfied), the auto merge is already failing 🤔 

![grafik](https://user-images.githubusercontent.com/100134454/180192061-9bb5500b-869c-4adb-ac95-bc68e0178f95.png)

In the screenshot the button already says "Merge pull request". I have the feeling that that is already wrong. Shouldn't that be anything like "prepare for automerge"?

**Comment by Celebrate-Reinhard** [comment]
Thank you, @izavads!

> also verify that atleast one check that you are running is required ( you can set it to be required in the protected branch rule ) .

This is exactly what I missed!

> i did not split the automerge part to be in a seperated yml workflow

I also reverted my split and it's now again like desribed [yesterday](https://github.com/peter-evans/enable-pull-request-automerge/issues/355#issuecomment-1190324241).

Details for everyone with the same issue like me.

This is how it should look like:
![grafik](https://user-images.githubusercontent.com/100134454/180236085-7266adfa-d800-41b3-9b63-10bd0c9e6285.png)

I just had to define the CI jobs as "required before merging"
![grafik](https://user-images.githubusercontent.com/100134454/180238433-79d8f29a-3d10-4b8c-a4eb-2c938d9b9c63.png)

Before, the CI jobs ran and they even could fail, but they were not required. So the PR was always in a merge ready state from the very beginning. And being  in that state the auto-merge flag cannot be set; not by UI and not by @peter-evans 's action.

**Comment by Celebrate-Reinhard** [comment]
@peter-evans suggestion for making [your docu](https://github.com/peter-evans/enable-pull-request-automerge#conditions) even clearer: 

> The pull request base must have a branch protection rule with at least one requirement enabled.
> In most cases a good candidate for this is a CI job with integration tests that runs for every PR anyway. Add that one in `Settings > Branches > Branch protection rule > Require status checks to pass before merging`

Thank you for your help and the great action you created!

**Comment by novotl** [comment]
In our case another repo admin removed the branch protection rule, so there was no required condition to wait for and it could be merged immediately. After adding back the required check all works fine 👍 .

**Comment by peter-evans** [comment]
Thanks @williamboman 

Released as `v2.2.0` / `v2`
